src/core/txs/wallet_pool.c

```diff
@@ -0,0 +1,131 @@
+#include <stdio.h>
+#include "wallet_pool.h"
+#include "sign_tx.h"
+
+void wallet_init() {
+  wallet_pool = NULL;
+  key_pool = NULL;
+}
+
+/* WALLET POOL FUNCS */
+
+WalletEntry *wallet_pool_add(
+    Transaction *tx, unsigned int vout, mbedtls_ecdsa_context *key_pair
```

Comment:
Not sure I track - is this in reference to using the UTXOPoolKey? If so, I think the current implementation makes since. The wallet pool is effectively a second UTXO pool specific to the node that also tracks keys. It should be updating whenever a new block is added, so it should save us from having to scan the entire chain for things?